pyiode/pyiode_model.pyx

```diff
@@ -6,45 +6,246 @@
+    >>> iode.model_calc_scc(nb_passes=1)        # doctest: +SKIP
     
+    '''
+    if isinstance(eqs_list, list):
+        eqs_list = ','.join(eqs_list)
+    elif eqs_list is None:
+        eqs_list = ""
+    
+    if IodeModelCalcSCC(nb_passes, 
+                         cstr(pre_listname), cstr(inter_listname), cstr(post_listname),
+                         cstr(eqs_list)):
+        raise RuntimeError(f"Cannot create the model Connex Components")
+    
+
+def model_simulate_scc( sample_from: str, sample_to: str, 
+                        pre_listname: str = "_PRE", 
+                        inter_listname: str = "_INTER", 
+                        post_listname: str = "_POST",
+                        eps: float = 0.0001, 
+                        relax: float = 1.0, 
+                        maxit: int = 100, 
+                        init_values: int = KV_INIT_TM1, 
+                        debug: int = 0, 
+                        newton_eps: float = 1e-6, 
+                        newton_maxit: int = 50, 
+                        newton_debug: int = 0):
+    '''
+    Simulate a model defined by the 3 Connex Components (stored in 3 lists).
+    '''
+
+    model_simulate_save_parms(eps, relax, maxit, init_values, -1, -1)
+    
+    if IodeModelSimulateSCC(cstr(sample_from), cstr(sample_to), 
+                         cstr(pre_listname), cstr(inter_listname), cstr(post_listname),
+                         eps, relax, maxit, init_values, debug, 
+                         newton_eps, newton_maxit, newton_debug):
+        raise RuntimeError(f"model_simulate_scc() failed")
+        
+
+def model_simulate_save_parms(
+                    eps: float = 0.0001, 
+                    relax: float = 1.0, 
+                    maxit: int = 100, 
+                    init_values: int = KV_INIT_TM1, 
+                    sort_algo: int = SORT_BOTH, 
+                    nb_passes: int = 1):
+    ''' 
+        Save in the global variabels KSIM_* the simulation parameters used during the last 
+        call to model_simulate() and model_simulate_scc(). 
+        The purpose of this function is to enable later reporting
+        via the functions model_simulate_maxit(), model_simulate_eps()...
+    '''
+    global KSIM_MAXIT
+    global KSIM_EPS
+    global KSIM_RELAX
+    global KSIM_PASSES
+    global KSIM_SORT
+    global KSIM_START
+
+    KSIM_MAXIT = maxit
+    KSIM_EPS = eps
+    KSIM_RELAX = relax
+    KSIM_START = init_values
+
+    if nb_passes >= 0: KSIM_PASSES = nb_passes  # not used by model_simulate_scc()
+    if sort_algo >= 0: KSIM_SORT = sort_algo    # id.
+    
+   
+def model_simulate_maxit() -> int:
+    ''' Returns the maxit parameter of the last simulation '''
+    return KSIM_MAXIT
+
+def model_simulate_eps() -> float:
+    ''' Returns the eps parameter of the last simulation '''
+    return KSIM_EPS
+
+def model_simulate_relax() -> float:
+    ''' Returns the relax parameter of the last simulation '''
+    return KSIM_RELAX
```

Comment:
return __KSIM_RELAX

Reply:
As explained above, `KSIM*` come from the C library and must not be renamed.